smtpserver/server.go

```diff
@@ -2418,10 +2418,15 @@ func (c *conn) deliver(ctx context.Context, recvHdrFor func(string) string, msgW
 					m.MessageID = messageid
 					m.MessageHash = messagehash
 					acc.WithWLock(func() {
-						if hasSpace, err := acc.TidyRejectsMailbox(c.log, conf.RejectsMailbox); err != nil {
-							log.Errorx("tidying rejects mailbox", err)
-						} else if hasSpace {
-							if err := acc.DeliverMailbox(log, conf.RejectsMailbox, m, dataFile, false); err != nil {
+						hasSpace := true
+						var err error = nil
+						if !conf.KeepRejects {
+							if hasSpace, err = acc.TidyRejectsMailbox(c.log, conf.RejectsMailbox); err != nil {
```

Comment:
could you move the err != nil check to below outside the "if !conf.KeepRejects" check? so only the TidyRejectsMailbox in the "if !conf.KeepRejects" check. the err-checking logic will stay closer to what it was. otherwise, on error from TidyRejectsMailbox, with the change as is, we will start logging "not storing spammy mail to full rejects mailbox" as well as the error.
looks good otherwise, thanks for the PR!

Reply:
Done!